src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -389,10 +420,22 @@ public boolean getLocationPermissions() {
 	@Override
 	public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
 		super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-		if (requestCode != ACCESS_FINE_LOCATION_PERMISSION_REQUEST_CODE) {
+		boolean granted = grantResults.length > 0 && grantResults[0] == PERMISSION_GRANTED;
+
+		if (requestCode == ACCESS_FINE_LOCATION_PERMISSION_REQUEST_CODE) {
```

Comment:
The recommended practice is to check if the permission is granted or denied ([article](https://developer.android.com/training/permissions/requesting)), otherwise is like not considering the user decision (as tested in Samsung Android 11). 
We'll be handling this for Location + some refactoring in this Activity in another ticket because of the location testing.

Reply:
Ticket: https://github.com/medic/cht-android/issues/189